pkg/apis/klum.cattle.io/v1alpha1/types.go

```diff
@@ -21,10 +21,22 @@ type User struct {
 	Status            UserStatus `json:"status,omitempty"`
 }
 
+type GithubSyncSpec struct {
+	Owner       string `json:"owner"`
+	Repository  string `json:"repository"`
+	Environment string `json:"environment,omitempty"`
+	SecretName  string `json:"secret_name"`
+}
+
+type SyncSpec struct {
+	Github GithubSyncSpec `json:"github,omitempty"`
```

Comment:
Suggest specifying this as pointer, so that `null` is a valid value

Then do `!= nil` checks to see if you should consider the GitHub sync

Reply:
Done in [699d745](https://github.com/jadolg/klum/pull/19/commits/699d745e7ea069f3986e1a2a5f046b154803fab3)